### PR TITLE
Blockly hoc2015 updates

### PR DESCRIPTION
**Pull request by cpirich**
- Updates blockly script through level 9
- Added addPoints, removePoints, whenTouchGoal
- renamed whenTouchItem -> whenTouchCharacter (matches other script)
- renamed whenTouchWall -> whenTouchObstacle (matches other script)
- fix `autoArrowSteer` for blockly levels (synchronously execute with `callCmd` since there is no `cmdQueue` when the move commands are inserted)
- fix `whenTouchGoal` for blockly levels (must call `executeQueue()`)

### CODE REVIEW

**Comment by cpirich** [comment]
@breville @islemaster

File: apps/i18n/studio/en_us.json [hunk 1]
```diff
@@ -2,6 +2,12 @@
   "actor": "actor",
   "addCharacter": "add a",
   "addCharacterTooltip": "Add a character to the scene.",
+  "addPoints10": "add 10 points",
```
Comment (breville) [inline apps/i18n/studio/en_us.json:5]: not using a number field?

File: apps/src/studio/studio.js [hunk 2]
```diff
@@ -1040,6 +1040,10 @@ Studio.onTick = function() {
     Studio.succeededTime = currentTime;
   }
 
+  if (!animationOnlyFrame) {
+    Studio.executeQueue('whenTouchGoal');
```
Comment (breville) [inline apps/src/studio/studio.js:1044]: i never pondered multiple queues before.. we execute them all separately when it's their time?

Comment (breville) [inline apps/src/studio/studio.js:1044]: more specifically, i see that we call handler for this event, and callHandler has this in the blockly codepath:

// Note: we skip executing the code if we have not completed executing
// the cmdQueue on this handler (checking for non-zero length)

does this mean that the 'whenTouchGoal' cmdQueue can be already have things in it and that's why it doesn't get called in blockly, or is there another reason callHandler isn't enough?

File: apps/i18n/studio/en_us.json (hunk 1 above)
Comment (cpirich) [inline apps/i18n/studio/en_us.json:5]: blockly determines which part of the block should be a combo box - this approach is very flexible for localization. See `moveDistance25` etc. for another example

Comment (cpirich) [inline apps/i18n/studio/en_us.json:5]: and, yes, we typically use combo boxes in blockly. we do have `params` version of some blocks in playlab, but they make the student work a bit more complex. This seemed like a good case to keep it simple. We can always add another `params` version of this block later.

File: apps/src/studio/studio.js (hunk 2 above)
Comment (cpirich) [inline apps/src/studio/studio.js:1044]: in playlab, `callHandler` will execute the generated code, but all playlab generated code simply adds commands to queues. Those queues need to be executed separately. We typically call `executeQueue` soon after we potentially add something to a queue.

The code you mention will occasionally skip adding commands to the queue (depends on the `allowQueueExtension` param). For simple things like `whenUp`, there really isn't much point in adding multiple events in the queue (this is like spoiling extra repaint messages in the desktop world). But for other cases, (collisions with items/projectiles), we will queue up multiple commands before executing them.

**Pull request merged**